src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java

```diff
@@ -113,6 +135,20 @@ private CaseDetails verifyReadAccess(CaseType caseType, Set<String> userRoles, C
         return caseDetails;
     }
 
+    private Set<String> getCaseRoles(String caseId) {
+        return caseUserRepository
+            .findCaseRoles(Long.valueOf(caseId), userRepository.getUserId())
+            .stream()
+            .collect(Collectors.toSet());
```

Comment:
This logic seems to be duplicated 3 times in this PR, might be good to extract it into the case role service.

Reply:
👍